homeassistant/components/hue/__init__.py

```diff
@@ -1,79 +1,41 @@
 """Support for the Philips Hue system."""
-import asyncio
-import logging
 
 from aiohue.util import normalize_bridge_id
-import voluptuous as vol
 
 from homeassistant import config_entries, core
 from homeassistant.components import persistent_notification
-from homeassistant.helpers import config_validation as cv, device_registry as dr
-from homeassistant.helpers.service import verify_domain_control
+from homeassistant.helpers import device_registry as dr
 
 from .bridge import HueBridge
-from .const import (
-    ATTR_GROUP_NAME,
-    ATTR_SCENE_NAME,
-    ATTR_TRANSITION,
-    CONF_ALLOW_HUE_GROUPS,
-    CONF_ALLOW_UNREACHABLE,
-    DEFAULT_ALLOW_HUE_GROUPS,
-    DEFAULT_ALLOW_UNREACHABLE,
-    DOMAIN,
-)
-
-_LOGGER = logging.getLogger(__name__)
-SERVICE_HUE_SCENE = "hue_activate_scene"
+from .const import DOMAIN, SERVICE_HUE_ACTIVATE_SCENE
+from .migration import check_migration
+from .services import async_register_services
 
 
 async def async_setup_entry(
     hass: core.HomeAssistant, entry: config_entries.ConfigEntry
-):
+) -> bool:
     """Set up a bridge from a config entry."""
+    # check (and run) migrations if needed
+    await check_migration(hass, entry)
 
-    # Migrate allow_unreachable from config entry data to config entry options
-    if (
-        CONF_ALLOW_UNREACHABLE not in entry.options
-        and CONF_ALLOW_UNREACHABLE in entry.data
-        and entry.data[CONF_ALLOW_UNREACHABLE] != DEFAULT_ALLOW_UNREACHABLE
-    ):
-        options = {
-            **entry.options,
-            CONF_ALLOW_UNREACHABLE: entry.data[CONF_ALLOW_UNREACHABLE],
-        }
-        data = entry.data.copy()
-        data.pop(CONF_ALLOW_UNREACHABLE)
-        hass.config_entries.async_update_entry(entry, data=data, options=options)
-
-    # Migrate allow_hue_groups from config entry data to config entry options
-    if (
-        CONF_ALLOW_HUE_GROUPS not in entry.options
-        and CONF_ALLOW_HUE_GROUPS in entry.data
-        and entry.data[CONF_ALLOW_HUE_GROUPS] != DEFAULT_ALLOW_HUE_GROUPS
-    ):
-        options = {
-            **entry.options,
-            CONF_ALLOW_HUE_GROUPS: entry.data[CONF_ALLOW_HUE_GROUPS],
-        }
-        data = entry.data.copy()
-        data.pop(CONF_ALLOW_HUE_GROUPS)
-        hass.config_entries.async_update_entry(entry, data=data, options=options)
-
+    # setup the bridge instance
     bridge = HueBridge(hass, entry)
-
-    if not await bridge.async_setup():
+    if not await bridge.async_initialize_bridge():
         return False
```

Comment:
Should we return `ConfigEntryNotReady` so we retry setup if it's due to a connection error ?

Reply:
Yeah, but that's already handled in the async_initialize_bridge method itself.
If the connection is not yet ready the ConfigEntryNotReady will be raised.

In fact the whole return bool is a bit unneeded as it will either initialize or raise

Reply:
Oh no, the logic is correct now. Returns false if the setup didn't succeed.
Raises NotReady if if's not ready. So it is already good as it is now. At least, from my perspective.